plugins/TrollManagement/tests/TrollManagementTest.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * @author David Barbier<david.barbier@vanillaforums.com>
+ * @copyright 2009-2021 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace VanillaTests\TrollManagement;
+
+use Garden\Container\Container;
+use UserModel;
+use RoleModel;
+use BanModel;
+use VanillaTests\APIv0\TestDispatcher;
+use VanillaTests\SetupTraitsTrait;
+use VanillaTests\SiteTestCase;
+use VanillaTests\TestInstallModel;
+
+/**
+ * Class TrollManagementTest
+ */
+class TrollManagementTest extends SiteTestCase {
+    use SetupTraitsTrait;
+
+    /** @var UserModel */
+    protected $userModel;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setUp(): void {
+        parent::setUp();
+        $this->createUserFixtures();
+    }
+
+    /**
+     * Setup routine, run before the test class is instantiated.
+     */
+    public static function setupBeforeClass(): void {
+        self::$addons = ['vanilla', 'trollmanagement'];
+
+        parent::setupBeforeClass();
+    }
+
+    /**
+     * Tests the automatic assignment of the "applicant" status to a new user registration if the maximum amount of
+     * user accounts sharing the same fingerprint is reached.
+     */
+    public function testRegisterApplicant(): void {
+        /** @var \Gdn_Configuration $configuration */
+        $configuration = static::container()->get('Config');
+
+        $configuration->set('TrollManagement.PerFingerPrint.Enabled', true);
+        $configuration->set('TrollManagement.PerFingerPrint.MaxUserAccounts', 3);
+
+        // Ensure all future registered dummy users uses the same fingerprint.
+        $_COOKIE['__vnf'] = 'THISISAFAKEFINGERPRINT';
```

Comment:
For the sake of tidying up the environment, this should probably be unset as part of the test case's `teardown`.

Reply:
Agreed. Done.